### DISCUSSION TITLE
How to generate a report for the coverage of tests and doctests combined?

### DISCUSSION THREAD

**Discussion by ISibboI** [xd009642/tarpaulin#1426] [Q&A] [answered]
It seems like `cargo-tarpaulin` can only run all non-doc tests, or all doctests, but not both combined. How can I generate a coverage report that takes both normal tests and doctests into account?

**Comment by xd009642** [comment, marked as answer]
`cargo tarpaulin --run-types tests --run-types doctests` or `cargo tarpaulin --doc --tests` would do both. Test types are additive.

**Reply by ISibboI** [reply]
I understand thanks. I was confused because --run-types was changed to allow only one argument, and the documentation of --doc specifies that it runs only doctests.

**Reply by xd009642** [reply]
That's because of clap changes, you now have to do it multiple times like `--run-types A --run-types B` bit unfortunate but staying on clap 3 wasn't great either

**Comment by ISibboI** [comment]
I see. Thank you a lot for the swift reply!
